lib/luahid.c

```diff
@@ -0,0 +1,102 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 Jieming Zhou <qrsikno@gmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/version.h>
+#include <linux/slab.h>
+#include <linux/limits.h>
+#include <asm/byteorder.h>
+
+#include <linux/hid.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+
+#include <lunatik.h>
+
+typedef struct luahid_s {
+	lunatik_object_t *runtime;
+	struct hid_driver driver;
+} luahid_t;
+
+static const struct hid_device_id hid_table[] = {
```

Comment:
please, prefix *everything" with `luahid_`; let's keep your namespace clear, so we can spot easily what was defined by your module, what's lunatik, what's lua, what's kernel..
```suggestion
static const struct hid_device_id luahid_table[] = {
```
```suggestion
static const struct hid_device_id hid_table[] = {
```
```suggestion
static const struct hid_device_id hid_table[] = {
```
```suggestion
static const struct hid_device_id hid_table[] = {
```

Reply:
done.